tools/gen-manifest.js

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env node
+'use strict';
+
+const crypto = require('crypto');
+const fs = require('fs');
+const { format } = require('util');
+
+let outfile = process.argv[2];
+let appinfo = JSON.parse(fs.readFileSync('appinfo.json'));
+let ipkfile = format('%s_%s_all.ipk',appinfo.id, appinfo.version);
```

Comment:
Space missing here.

Also it's the first time I've seen someone use `util.format`, in practice everyone just ends up with template literals like this:
```
`${appinfo.id}_${appinfo.version}_all.ipk`
```

(don't need to fix this, just a tip on more common syntax)

Reply:
I would also prefer string interpolation, ~~but compiler is not happy with that~~ turns out it works! Changed :)